<?xml version="1.0" ?>
<robot name="omy_f3m">

  <ros2_control name="GazeboSystem" type="system">
    <hardware>
      <plugin>gz_ros2_control/GazeboSimSystem</plugin>
    </hardware>
    
    <joint name="joint1">
      <command_interface name="position"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
    </joint>

    <joint name="joint2">
      <command_interface name="position"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
    </joint>

    <joint name="joint3">
      <command_interface name="position"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
    </joint>

    <joint name="joint4">
      <command_interface name="position"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
    </joint>

    <joint name="joint5">
      <command_interface name="position"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
    </joint>

    <joint name="joint6">
      <command_interface name="position"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
    </joint>
  </ros2_control>

  <gazebo>
    <plugin filename="gz_ros2_control-system" name="gz_ros2_control::GazeboSimROS2ControlPlugin">
      <parameters>$(find robotis_gz_sim)/config/controllers.yaml</parameters>
    </plugin>
  </gazebo>

  <link name="world"/>
  
  <joint name="world_fixed" type="fixed">
    <parent link="world"/>
    <child link="link0"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>

  <link name="link0">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/base_unit.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"><color rgba="0.2 0.2 0.2 1"/></material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/base_unit.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="-4.02e-03 1.30e-03 5.50e-02"/>
      <mass value="1.52"/>
      <inertia ixx="0.0046" ixy="0.0" ixz="0.0" iyy="0.0049" iyz="0.0" izz="0.0064"/>
    </inertial>
  </link>

  <joint name="joint1" type="revolute">
    <parent link="link0"/>
    <child link="link1"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.1715"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-6.28" upper="6.28" velocity="4.8"/>
    <dynamics damping="0.05"/>
  </joint>

  <link name="link1">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/link1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"><color rgba="0.2 0.2 0.2 1"/></material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/link1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <inertial>
       <mass value="2.06"/>
       <inertia ixx="0.003" ixy="0.0" ixz="0.0" iyy="0.0027" iyz="0.0" izz="0.0022"/>
    </inertial>
  </link>

  <joint name="joint2" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin rpy="-1.5707 0 0" xyz="0 -0.1215 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-6.28" upper="6.28" velocity="4.8"/>
    <dynamics damping="0.1"/>
  </joint>

  <link name="link2">
    <visual>
      <origin rpy="1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/link2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"><color rgba="0.2 0.2 0.2 1"/></material>
    </visual>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/link2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.67"/>
      <inertia ixx="0.051" ixy="0.0" ixz="0.0" iyy="0.050" iyz="0.0" izz="0.0045"/>
    </inertial>
  </link>

  <joint name="joint3" type="revolute">
    <parent link="link2"/>
    <child link="link3"/>
    <origin rpy="0 0 0" xyz="0 -0.247 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-6.28" upper="6.28" velocity="4.8"/>
    <dynamics damping="0.1"/>
  </joint>

  <link name="link3">
    <visual>
      <origin rpy="1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/link3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"><color rgba="0.2 0.2 0.2 1"/></material>
    </visual>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/link3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="2.38"/>
      <inertia ixx="0.022" ixy="0.0" ixz="0.0" iyy="0.021" iyz="0.0" izz="0.0027"/>
    </inertial>
  </link>

  <joint name="joint4" type="revolute">
    <parent link="link3"/>
    <child link="link4"/>
    <origin rpy="0 0 0" xyz="0 -0.2195 0.1215"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-6.28" upper="6.28" velocity="4.8"/>
    <dynamics damping="0.1"/>
  </joint>

  <link name="link4">
    <visual>
      <origin rpy="1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/link4.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"><color rgba="0.2 0.2 0.2 1"/></material>
    </visual>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/link4.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.40"/>
      <inertia ixx="0.0017" ixy="0.0" ixz="0.0" iyy="0.0015" iyz="0.0" izz="0.0012"/>
    </inertial>
  </link>

  <joint name="joint5" type="revolute">
    <parent link="link4"/>
    <child link="link5"/>
    <origin rpy="1.5707 0 0" xyz="0 0 -0.113"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-6.28" upper="6.28" velocity="4.8"/>
    <dynamics damping="0.1"/>
  </joint>

  <link name="link5">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/link5.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"><color rgba="0.2 0.2 0.2 1"/></material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/link5.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.40"/>
      <inertia ixx="0.0017" ixy="0.0" ixz="0.0" iyy="0.0012" iyz="0.0" izz="0.0015"/>
    </inertial>
  </link>

  <joint name="joint6" type="revolute">
    <parent link="link5"/>
    <child link="link6"/>
    <origin rpy="-1.5708 0 0" xyz="0 0 0.1155"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-6.28" upper="6.28" velocity="4.8"/>
    <dynamics damping="0.1"/>
  </joint>

  <link name="link6">
    <visual>
      <origin rpy="1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/link6.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"><color rgba="0.2 0.2 0.2 1"/></material>
    </visual>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotis_gz_sim/meshes/omy_f3m/link6.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.40"/>
      <inertia ixx="0.0005" ixy="0.0" ixz="0.0" iyy="0.0006" iyz="0.0" izz="0.00019"/>
    </inertial>
  </link>
</robot>